Marlin/src/feature/mmu/mmu2.cpp

```diff
@@ -508,9 +508,31 @@ static void mmu2_not_responding() {
 
     set_runout_valid(false);
 
-    if (index != extruder) {
+    if (index != extruder)
+    {
+      #if ENABLED(MMU_IR_UNLOAD_MOVE)
+        if (FILAMENT_PRESENT()) {
+          DEBUG_ECHOLNPGM("Unloading\n");
+          //filament_ramming(); //unloading from printer side
+          ENABLE_AXIS_E0();
+          while (FILAMENT_PRESENT())
```

Comment:
This and the other loop could continue forever if the filament is jammed. It should have a limit so that if the unload fails correct action can occur.

What is the correct action? I'm not sure...perhaps trigger filament runout?

Reply:
It looks like the `manage_response` probably exists for the purpose of pausing if something goes wrong. Your loops probably need to define a maximum retract distance. If that is exceeded you would call `manage_response` with the expected arguments to allow the user to clear the issue.

Reply:
A maximum distance can be enforced for these loops using a simple countdown, based on whatever maximum length is determined for each case. I'll leave it to @niccoreyes to determine the appropriate limits.